lib/docker/util.rb

```diff
@@ -142,10 +144,33 @@ def create_dir_tar(directory)
     File.new(tempfile.path, 'r')
   end
 
+
+  # return the set of files that form the docker context
+  # implement this logic https://docs.docker.com/engine/reference/builder/#dockerignore-file
+  def docker_context(directory)
+    all_files = glob_all_files(File.join(directory, "**/*"))
+    dockerignore = File.join(directory, '.dockerignore')
+    return all_files unless all_files.include?(dockerignore)
+
+    # iterate over non-comments and non-blanks from the dockerignore file,
+    # starting with the initial glob as working set
+    File.read(dockerignore).each_line.map(&:strip)
```

Comment:
Can we move each of these to new lines? Would prefer if we're chaining vertically, to chain it all that way.